src/libraries/System.Net.Security/src/System/Net/Security/SslAuthenticationOptions.cs

```diff
@@ -119,8 +120,10 @@ internal void UpdateOptions(SslServerAuthenticationOptions sslServerAuthenticati
 
                 if (certificateWithKey != null && certificateWithKey.HasPrivateKey)
                 {
+                    bool ocspFetch = false;
+                    _ = AppContext.TryGetSwitch(EnableOcspStaplingContextSwitchName, out ocspFetch);
                     // given cert is X509Certificate2 with key. We can use it directly.
-                    CertificateContext = SslStreamCertificateContext.Create(certificateWithKey, additionalCertificates: null, offline: false, trust: null, noOcspFetch: true);
+                    CertificateContext = SslStreamCertificateContext.Create(certificateWithKey, additionalCertificates: null, offline: false, trust: null, noOcspFetch: !ocspFetch);
```

Comment:
are we changing behavior here? I thought https://github.com/dotnet/runtime/issues/96395. was only about ability to control it.

Reply:
We're leaving the behavior as-is, unless AppContext switch has been used.